### PR TITLE
Docker: Make SFTP key auth easy, and adding plugins/themes less likely to break Backup/Scan

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

* Adds a `docker/data/ssh.keys/`, into which ssh public key files can be placed to allow key-based login for SFTP.
* Adds a `docker/compose-volumes.yml` file for defining the bind mounts for WordPress and its plugins and themes, so these won't get out of sync between the web service container and the sftp container.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

I'll assume you're already reasonably familiar with the docker-based dev environment.

* `yarn docker:down`
* Generate an ssh keypair, and copy the public key into `docker/data/ssh.keys/`. Filename doesn't matter as long as it ends with `.pub`.
* `yarn docker:compose-volumes` if necessary to generate a default `docker/compose-volumes.yml`. Or copy `docker/compose-volumes.yml.sample` manually.
* Add a plugin or theme by editing `docker/compose-volumes.yml` following the instructions in comments in that file.
* `yarn docker:up`
* Verify your plugin or theme is available in WordPress's UI.
* Connect to the container using sftp and the private key generated earlier.
    * On a command line, that'd probably be something like `sftp -o IdentitiesOnly=yes -i /path/to/key -P 1022 wordpress@localhost`.
* When connected, see that your added plugin or theme can be found via sftp.

#### Proposed changelog entry for your changes:
* I see previous changes to the Docker env haven't done changelog entries, so I guess this won't either

#### Note
* This is my first attempt at a PR, please let me know if I did anything wrong!

### CODE REVIEW

File: docker/compose-extras.yml.sample [hunk 1]
```diff
@@ -1,5 +1,3 @@
 version: '3.3'
 #services:
 #  wordpress:
-#    volumes:
```
Comment (jeherve) [inline docker/compose-extras.yml.sample:4]: Do you think we should add a comment here pointing to the new solution instead?

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -46,10 +46,10 @@ services:
     container_name: jetpack_sftp
     image: atmoz/sftp
     volumes:
-      - ..:/home/wordpress/var/www/html/wp-content/plugins/jetpack
+      ## WordPress and its plugins and themes should be defined in docker/compose-volumes.yml{.sample}, not here!
+      ## That avoids haing to define them both here and in the "wordpress" service.
```
Comment (jeherve) [inline docker/docker-compose.yml:50]:
```suggestion
      ## That avoids having to define them both here and in the "wordpress" service.
```

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -75,17 +75,13 @@ services:
       - maildev
     image: automattic/jetpack-wordpress-dev:latest
     volumes:
-      - ..:/var/www/html/wp-content/plugins/jetpack
-      ## Kludge for not having docker contain recursive stuff
-      ## You will see on your filesystem that this dir gets created
+      ## WordPress and its plugins and themes should be defined in docker/compose-volumes.yml{.sample}, not here!
+      ## That avoids haing to define them both here and in the "sftp" service.
```
Comment (jeherve) [inline docker/docker-compose.yml:79]:
```suggestion
      ## That avoids having to define them both here and in the "sftp" service.
```

**Review by jeherve — COMMENTED** [review]
I like this! Once this is merged, it may be worth a P2 post inviting folks to switch from the old to the new implementation.

File: docker/compose-extras.yml.sample (hunk 1 above)
Comment (anomiex) [reply docker/compose-extras.yml.sample:4]: Sure, we can do that.

File: docker/docker-compose.yml (hunk 2 above)
Comment (anomiex) [reply docker/docker-compose.yml:50]: Thanks! I'll squash this into the update I'm about to push for the other comments.

File: docker/docker-compose.yml (hunk 3 above)
Comment (anomiex) [reply docker/docker-compose.yml:79]: Thanks! I'll squash this into the update I'm about to push for the other comments.

**Review by jeherve — APPROVED** [review]
🌟 🚢

**Comment by jeherve** [comment]
r213201-wpcom

**Pull request merged**